pulpcore.fc

```diff
@@ -1,16 +1,21 @@
-/usr/lib/pulp/bin/gunicorn		--	gen_context(system_u:object_r:pulpcore_exec_t,s0)
+/etc/pulp(/.*)?			gen_context(system_u:object_r:pulpcore_etc_t,s0)
+
+/usr/bin/rq		--	gen_context(system_u:object_r:pulpcore_exec_t,s0)
+/usr/bin/gunicorn	--	gen_context(system_u:object_r:pulpcore_server_exec_t,s0)
+
+/usr/libexec/pulpcore/.*	--	gen_context(system_u:object_r:pulpcore_exec_t,s0)
+
+/usr/lib/pulp/bin/gunicorn	--	gen_context(system_u:object_r:pulpcore_server_exec_t,s0)
 /usr/lib/pulp/bin/rq		--	gen_context(system_u:object_r:pulpcore_exec_t,s0)
 
-/usr/local/lib/pulp/bin/gunicorn		--	gen_context(system_u:object_r:pulpcore_exec_t,s0)
+/usr/local/lib/pulp/bin/gunicorn	--	gen_context(system_u:object_r:pulpcore_server_exec_t,s0)
 /usr/local/lib/pulp/bin/rq		--	gen_context(system_u:object_r:pulpcore_exec_t,s0)
 
 /var/lib/pulp/artifact(/.*)? 	gen_context(system_u:object_r:pulpcore_var_lib_t,s0)
 /var/lib/pulp/assets(/.*)? 	gen_context(system_u:object_r:pulpcore_var_lib_t,s0)
 /var/lib/pulp/tmp(/.*)? 	gen_context(system_u:object_r:pulpcore_var_lib_t,s0)
 /var/lib/pulp/upload(/.*)? 	gen_context(system_u:object_r:pulpcore_var_lib_t,s0)
+/var/lib/pulp(/.*)? 			gen_context(system_u:object_r:pulpcore_var_lib_t,s0)
 
-/var/run/pulpcore*(/.*)? 	gen_context(system_u:object_r:pulpcore_var_run_t,s0)
-/var/run/pulpcore-api(/.*)? 	gen_context(system_u:object_r:pulpcore_var_run_t,s0)
-/var/run/pulpcore-content(/.*)? 	gen_context(system_u:object_r:pulpcore_var_run_t,s0)
-/var/run/pulpcore-resource-manager(/.*)? 	gen_context(system_u:object_r:pulpcore_var_run_t,s0)
-/var/run/pulpcore-resource-worker[0-9]*(/.*)? 	gen_context(system_u:object_r:pulpcore_var_run_t,s0)
+/var/run/pulpcore-content(/.*)? 	gen_context(system_u:object_r:pulpcore_server_var_run_t,s0)
+/var/run/pulpcore.*		 	gen_context(system_u:object_r:pulpcore_var_run_t,s0)
```

Comment:
I wonder: `/var/run` is now usually a symlink to `/run` so is that really relevant? At least in in https://github.com/theforeman/puppet-pulpcore/pull/116 (the Puppet based installer) I'm dropping the use of these directories altogether by dropping pid files. They're useless on systemd anyway.

Reply:
@ekohl, I don't understand. It's a true that /var/run is a symlink to /run, but we have equivalency rules for that. Systemd uses the runtime dirs as well as the actual services on the system I have access to.

Reply:
To be clear: I was suggesting that the systemd unit files should be modified to not use pid files anymore, removing the need for these run dirs. That's probably more something for the installer maintainers

Reply:
In my instance there are not only pid files, but also other files in /run/pulpcore-*/, so dropping the support right away means the tests will stop working. I don't have any other way how to verify the policy is sufficient.

Reply:
I'd suggest to keep it in the SELinux policy, but I'd ask @mikedep333 to move away from pid files in the Ansible installer so this is no longer needed.

Reply:
I am also for keeping it now, note there are a dozen of lines related to runtime files usage. Once there are some changes in the code or installer, the policy would need to be adjusted anyway, so we can drop these lines.